ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -1563,6 +1563,7 @@ class HelixClusterChangeHandler
     private final CountDownLatch routingTableInitLatch = new CountDownLatch(1);
     private final List<ClusterMapChangeListener> clusterMapChangeListeners = new ArrayList<>();
     private volatile boolean dataNodeConfigInitialized = false;
+    private volatile boolean dataNodeConfigInitializedForLocalNode = false;
```

Comment:
We don't actually need this new variable, we can just use `dataNodeConfigIntialized` variable. 
I would like to change the interface of the callback to `onDataNodeConfigChange(List<DataNodeConfig> configs)` so the listener can do whatever they want to do with the configs.